contracts/minter-suite/Minters/MinterSetPriceV5Merkle.sol

```diff
@@ -0,0 +1,684 @@
+    }
+
+    /**
+     * @notice Inactive function - requires Merkle proof to purchase.
+     */
+    function purchaseTo(
+        address,
+        uint256,
+        address
+    ) public payable returns (uint256) {
+        revert("Must provide Merkle proof");
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId`.
+     * @param _projectId Project ID to mint a token on.
+     * @param _coreContract Contract address of the core contract.
+     * @param _proof Merkle proof for the given project.
+     * @return tokenId Token ID of minted token
+     */
+    function purchase(
+        uint256 _projectId,
+        address _coreContract,
+        bytes32[] calldata _proof
+    ) external payable returns (uint256 tokenId) {
+        tokenId = purchaseTo_do6(
+            msg.sender,
+            _projectId,
+            _coreContract,
+            _proof,
+            address(0)
+        );
+        return tokenId;
+    }
+
+    /**
+     * @notice gas-optimized version of purchase(uint256).
+     */
+    function purchase_H4M(
+        uint256 _projectId,
+        address _coreContract,
+        bytes32[] calldata _proof
+    ) external payable returns (uint256 tokenId) {
+        tokenId = purchaseTo_do6(
+            msg.sender,
+            _projectId,
+            _coreContract,
+            _proof,
+            address(0)
+        );
+        return tokenId;
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId` and sets
+     * the token's owner to `_to`.
+     * @param _to Address to be the new token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _coreContract Contract address of the core contract.
+     * @param _proof Merkle proof for the given project.
+     * @return tokenId Token ID of minted token
+     */
+    function purchaseTo(
+        address _to,
+        uint256 _projectId,
+        address _coreContract,
+        bytes32[] calldata _proof
+    ) external payable returns (uint256 tokenId) {
+        return
+            purchaseTo_do6(_to, _projectId, _coreContract, _proof, address(0));
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId` and sets
+     * the token's owner to `_to`.
+     * @param _to Address to be the new token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _coreContract Contract address of the core contract.
+     * @param _proof Merkle proof for the given project.
+     * @return tokenId Token ID of minted token
+     */
+    function purchaseTo(
+        address _to,
+        uint256 _projectId,
+        address _coreContract,
+        bytes32[] calldata _proof,
+        address _vault
+    ) external payable returns (uint256 tokenId) {
+        return purchaseTo_do6(_to, _projectId, _coreContract, _proof, _vault);
+    }
+
+    /**
+     * @notice gas-optimized version of purchaseTo(address, uint256).
+     */
+    function purchaseTo_do6(
```

Comment:
since we haven't really ended up calling the gas-optimized function names during purchases, I would be in favor in removing them from these new minters if we all agree on that. Would make things a bit more readable.

Reply:
after discussion: let's remove these for now, but make a note for the future to potentially re-introduce when we have more ubiquitous usage of data client library